audbcards/core/dataset.py

```diff
@@ -279,6 +297,290 @@ def version_link(self) -> str:
+        if has_maximums:
+            header_line += ',Max'
+            columns.append('Max')
+        if has_labels:
+            header_line += ',Labels'
+            columns.append('Labels')
+        if has_mappings:
+            header_line += ',Mappings'
+            columns.append('Mappings')
+
+        scheme_info['header_line'] = header_line
+        scheme_info['has_minimums'] = has_minimums
+        scheme_info['has_maximums'] = has_maximums
+        scheme_info['has_labels'] = has_labels
+        scheme_info['has_mappings'] = has_mappings
+
+        scheme_info['columns'] = columns
+        return scheme_info
+
+    @property
+    def dataset_schemes(self) -> list:
+
+        db = self.header
+        dataset_schemes = []
+        for scheme_id in db.schemes:
+            dataset_scheme = self._scheme_to_list(scheme_id)
+            dataset_schemes.append(dataset_scheme)
+
+        cols = self.scheme_info['columns']
+        data = pd.DataFrame.from_dict(dataset_schemes)[cols]
+        filter = data.applymap(lambda d: d == [])
+        data.mask(filter, other='', inplace=True)
+        scheme_data = np.array(data).tolist()
+        return scheme_data
+
+
+class Datacard(object):
+
+    def __init__(self, dataset: Dataset):
+
+        self._dataset = dataset
+
+    @property
+    def props(self) -> dict:
+        props = self._dataset.properties()
+        return props
+
+    def _render_template(self):
+
+        t_dir = os.path.join(os.path.dirname(__file__), 'templates')
+        environment = jinja2.Environment(loader=jinja2.FileSystemLoader(t_dir),
+                                         trim_blocks=True)
+        # Provide Jinja filter access to Python build-ins/functions
+        environment.filters.update(
+            zip=zip,
+            tw=self._trim_trailing_whitespace,
+        )
+        template = environment.get_template("datacard.j2")
+        content = template.render(self.props)
+        return content
+
+    @staticmethod
+    def _trim_trailing_whitespace(x: list):
+        """J2 filter to get rid of trailing empty table entries within a row.
+
+        Trims last entry if present.
+
+        Args:
+            x: untrimmed single scheme table row
+        Returns:
+            trimmed single scheme table row
+        """
+
+        if x[-1] == '':
+            x.pop()
+
+        return x
+
+    def save(self, ofpath: str = None):
+        """Save content of rendered template to rst.
+
+        Args:
+            ofpath: filepath to save rendered template to
+        Returns:
+            None
+
+        if ofpath is specified, the directory must exist.
+        """
+
+        content = self._render_template()
+
+        if ofpath is None:
+            ofpath = f'datasets/{self._dataset.name}_from_template.rst'
+
+        with open(ofpath, mode="w", encoding="utf-8") as fp:
+            fp.write(content)
+            print(f"... wrote {ofpath}")
+
+
+def create_datacard_page_from_template(dataset: Dataset):
```

Comment:
I would not add an additional `create_datacard_page_from_template()`, but simply replace `create_datacard_page()` by it as it should be always created from the templates.

Reply:
`create_datacard_page_from_template()` was only implemented to parallel the old implementation. Imo this does more bad than good and I have removed it in ffac9df. This remove the need to call it as done in  https://github.com/audeering/audbcards/pull/13#issuecomment-1731530202.

Saying that, I would find this a better way to use the `Datacard` class:

```
for (name, version) in zip(names, versions):
    print(f'Parse {name}-{version}... ', end='', flush=True)
    dataset = audbcards.Dataset(name, version)
    dc = audbcards.Datacard(dataset)
    dc.save()
```

In other word, I fully agree that all `create_datacard_page*` wrappers are dispensable.

Note that for `create_datasets_page` still has the `_from_template` counterpart.

Reply:
Maybe we can even reduce it to

```python
for (name, version) in zip(names, versions):
    print(f'Parse {name}-{version}... ', end='', flush=True)
    dc = audbcards.Datacard(name, version)
    dc.save()
```

This would also raise the question if we need `Dataset` as part of the public API,
but maybe it is still useful for other tasks then creating a datacard?

Reply:
Yes, I would prefer to keep it public. I want to use `Dataset` to extract its attributes for reporting purposes (e.g. for questions like  "how many hours of data exist having scheme X").

Reply:
Yes, let's stay with your solution for now.